### PR TITLE
Ks/#732 cleanup show command

### PR DESCRIPTION
**Pull request by KSchopmeyer**
Refactor command connection show processor and add tests

### CODE REVIEW

**Review by andy-maier — APPROVED** [review]
Pending rebase on master once PR #741 is merged.

File: pywbemtools/pywbemcli/_context_obj.py [hunk 1]
```diff
@@ -281,7 +281,7 @@ def execute_cmd(self, cmd):
             self.spinner_start()
         if self.pdb:
             import pdb  # pylint: disable=import-outside-toplevel
-            pdb.set_trace()
+            pdb.set_trace()  # pylint: disable=no-member
```
Comment (andy-maier) [inline pywbemtools/pywbemcli/_context_obj.py:284]: This Pylint disable is no longer needed, since it now works in your new virtual env.

**Review by andy-maier — CHANGES_REQUESTED** [review]
See comments

**Comment by andy-maier** [comment]
I think this PR completely addresses issue #732 , but please double check.

File: pywbemtools/pywbemcli/_cmd_connection.py [hunk 2]
```diff
@@ -112,14 +112,16 @@ def connection_show(context, name, **options):
...
-    * If the NAME argument is specified, the connection definition with that
-      name from the connections file is shown.
+    * If the NAME argument is specified, the connection information with that
+      name from the connections file is displayed or the current connection
+      if it is the same name.
 
     * If the NAME argument is '?', the command presents a list of connection
       definitions from the connections file and prompts the user for
-      selecting one, which is then shown.
+      selecting one, which is then displayed if a connection file exists.
```
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_connection.py]: DONE

[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_connection.py]: DONE

File: pywbemtools/pywbemcli/_cmd_connection.py [hunk 3]
```diff
@@ -399,34 +401,47 @@ def get_current_connection_name(context):
...
+    Parameters:
 
+      name (:term:`string):
+        Name that will be validated against repository or Noe if the name
+        is to be picked from a selection presented
+
+      context (:class:`'~pywbemtools.pywbemcli._context_obj.ContextObj`):
+        The current ContextObj instance.
+
+      connections_repo (:class:`'~pywbemtools._context_obj.ContextObj`):
```
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_connection.py:419]: DONE

File: pywbemtools/pywbemcli/_cmd_connection.py [hunk 4]
```diff
@@ -399,34 +401,47 @@ def get_current_connection_name(context):
...
-        if name in connections:
+        if name in connections_repo:
             return name
         raise click.ClickException(
             'Connection definition "{0}" not found in connections file "{1}"'.
-            format(name, connections.connections_file))
+            format(name, connections_repo.connections_file))
 
-    conn_names = sorted(list(six.iterkeys(connections)))
+    conn_names = sorted(list(six.iterkeys(connections_repo)))
```
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_connection.py:444]: DONE

File: pywbemtools/pywbemcli/_context_obj.py (hunk 1 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_context_obj.py]: DONE all of these now removed.  I guess I must have missed a rebase after the update to the other pr

File: pywbemtools/pywbemcli/_cmd_connection.py [hunk 5]
```diff
@@ -399,34 +401,47 @@ def get_current_connection_name(context):
...
+def pick_connection(name, context, connections_repo):
     """
-    Raise exception with message that connections file does not exist.
-    """
-    raise click.ClickException(
-        'Connections file "{0}" does not exist'.
-        format(connections.connections_file))
+    If name is None use the interactive mode to select the connection from the
+    list of connections in the connections file. If the name is provided, it is
+    tested against the names in the connections file.  If it is not provided,
```
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_connection.py:408]: REMVOVED. Not needed.

File: pywbemtools/pywbemcli/_cmd_connection.py [hunk 6]
```diff
@@ -399,34 +401,47 @@ def get_current_connection_name(context):
...
+
+    Parameters:
 
+      name (:term:`string):
+        Name that will be validated against repository or Noe if the name
+        is to be picked from a selection presented
```
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_cmd_connection.py:414]: DONE

File: pywbemtools/pywbemcli/_context_obj.py (hunk 1 above)
Comment (KSchopmeyer) [reply pywbemtools/pywbemcli/_context_obj.py]: I later figured out that the pylint issue is caused when pdb++ is installed and filed an issue.

**Pull request merged**